signup/signup.js

```diff
@@ -1,43 +1,128 @@
     handleInputFocusout(e.target, $errorMessageList);
   })
 );
+//focusout 콜백 함수
+const handleInputFocusout = ($target, $errorMessageList) => {
+  const { value, name } = $target;
+  const invalidKey = formValidator.findIvnalidKey(name, value);
+  errorMessageStatus($errorMessageList, $target, invalidKey);
+};
 
-$form.addEventListener("submit", function (e) {
-  let isFormValid = true;
-  let invalidKeylist = [true, true, true];
-  let counter = 0;
-  $inputList.forEach(($input) => {
-    const { value, name } = $input;
-    const invalidKey = formValidator.findIvnalidKey(name, value);
-    invalidKeylist[counter] = Boolean(invalidKey);
+//singup logic
+$form.addEventListener("submit", async function (e) {
+  e.preventDefault();
+  const dataSuccess = await postData();
 
-    counter++;
+  if (!invalidKeylist.includes(true)) {
+    $inputList.forEach(($input, i) => {
+      const { value, name } = $input;
+      const invalidKey = formValidator.findIvnalidKey(name, value);
+      invalidKeylist[i] = Boolean(invalidKey);
 
-    errorMessageStatus($errorMessageList, $input, invalidKey);
+      errorMessageStatus($errorMessageList, $input, invalidKey);
 
-    // 유효하지 않은 입력이 발견되면, isFormValid를 false로 설정합니다.
-    if (invalidKey) {
-      isFormValid = false;
+      // 유효하지 않은 입력이 발견되면, isFormValid를 false로 설정
+      if (invalidKey) {
+        isFormValid = false;
+      }
+    });
+  } else {
+    if (dataSuccess && okToSubmit) {
+      $form.submit();
     }
-  });
+  }
+});
 
-  // 폼이 유효하지 않으면 제출을 방지합니다.
-  if (invalidKeylist.includes(true)) {
+//check email
+//조금 불필요한 코드인 것 같기도 한데... 혹시 이 코드를 다른 곳에 합쳐서 없애버리는 게 더 나을까요?
+let okToSubmit = false;
+$inputList[0].addEventListener("keydown", async function (e) {
+  if (e.key === "Enter") {
     e.preventDefault();
+    const emailOk = await checkDuplicateEmail();
+    if (emailOk) {
+      $errorMessageList[0].textContent = "";
+      okToSubmit = true;
+    }
   }
 });
+
+const postData = async () => {
+  try {
+    let emailInput = $inputList[0].value;
+    let passwordInput = $inputList[1].value;
+
+    const data = JSON.stringify({
+      email: emailInput,
+      password: passwordInput,
+    });
+
+    const validateForm = await fetch(
+      "https://bootcamp-api.codeit.kr/api/sign-up",
+      {
+        method: "POST",
+        body: data,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    );
+
+    if (validateForm.status === 400) {
+      throw new Error("invalidForm");
+    } else if (!validateForm.ok) {
+      throw new Error("neworkError");
+    }
+    // alert("success");
+    // const responsedData = validateForm.data;
+  } catch (error) {
+    if (error.message === "invalidForm") alert("회원가입에 실패했습니다.");
+    else alert("네트워크 에러가 발생했습니다.");
+  }
+};
+
+const checkDuplicateEmail = async () => {
+  try {
+    let emailInput = $inputList[0].value;
+    const data = JSON.stringify({ email: emailInput });
+    const dupplicatedEmail = await fetch(
```

Comment:
오타 방지용 익스텐션 하나 소개시켜드리겠습니다

https://lawner.tistory.com/12

Reply:
또! 보통 fetch를 한 이후의 변수명은 `response`라고 명명하는 편입니다 

```suggestion
    const response = await fetch(어쩌구, 저쩌구)
    const XXX = await response.json();
```